base/stream.jl

```diff
@@ -876,6 +894,14 @@ function stop_reading(stream::LibuvStream)
     end
 end
 
+function readbytes!(s::LibuvStream, b::AbstractArray{UInt8}, nb=length(b))
+    wait_readnb(s, nb)
+    nr = nb_available(s)
+    resize!(b, nr) # shrink to just contain input data if was resized
```

Comment:
And this line won't work if you try to give it an array that shares data, `cannot resize array with shared data` - MbedTLS.jl does this via `pointer_to_array` with an array that is managed by the C library.